e2e/_suites/fleet/services.go

```diff
@@ -247,36 +175,42 @@ func newCentosInstaller(image string, tag string, version string) (ElasticAgentI
 		return runElasticAgentCommand(profile, image, service, ElasticAgentProcessName, "enroll", buildEnrollmentFlags(token))
 	}
 
-	binDir := "/var/lib/elastic-agent/data/elastic-agent-%s/"
+	workingDir := "/var/lib/elastic-agent"
+	binDir := workingDir + "/data/elastic-agent-%s/"
+	commitFileName := ".elastic-agent.active.commit"
+	homeDir := "/etc/elastic-agent/"
+
+	commitFile := homeDir + commitFileName
+
+	logsDir := binDir + "logs/"
+	logFileName := "elastic-agent-json.log"
```

Comment:
will this handle log rotations in between restarts? we will need also `*.1 ... *.N` files probably

Reply:
Does it mean that with every host/container restart, the log will rotate?

Reply:
I'm thinking about using `journalctl -u elastic-agent` instead of `cat`, which I guess it would hugely simplify the log output, but I'm seeing differences between both logs, more specifically about the **managed** modes:

**journalctl -u elastic-agent** (_using local mode_)
```
-- Logs begin at Tue 2021-03-02 09:24:31 UTC, end at Tue 2021-03-02 09:44:09 UTC. --
Mar 02 09:24:40 d94c90c8f079 systemd[1]: Started Elastic Agent is a unified agent to observe, monitor and protect your system..
Mar 02 09:24:40 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:40.984Z	INFO	warn/warn.go:18	The Elastic Agent is currently in BETA and should not be used in production
Mar 02 09:24:40 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:40.989Z	INFO	application/application.go:65	Detecting execution mode
Mar 02 09:24:40 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:40.991Z	INFO	application/application.go:74	Agent is managed locally
Mar 02 09:24:40 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:40.991Z	INFO	capabilities/capabilities.go:59	capabilities file not found in /opt/Elastic/Agent/capabilities.yml
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.391Z	INFO	[composable]	composable/controller.go:44	EXPERIMENTAL - Inputs with variables are currently experimental and should not be used in production
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.397Z	INFO	[composable.providers.docker]	docker/docker.go:43	Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.405Z	INFO	[api]	api/server.go:62	Starting stats endpoint
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.405Z	INFO	application/local_mode.go:165	Agent is starting
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.406Z	INFO	[api]	api/server.go:64	Metrics endpoint listening on: /tmp/elastic-agent/elastic-agent.sock (configured: unix:///tmp/elastic-agent/elastic-agent.sock)
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.411Z	INFO	application/local_mode.go:175	Agent is stopped
Mar 02 09:24:42 d94c90c8f079 elastic-agent[81]: 2021-03-02T09:24:42.411Z	INFO	application/periodic.go:76	Configuration changes detected
Mar 02 09:24:47 d94c90c8f079 systemd[1]: Stopping Elastic Agent is a unified agent to observe, monitor and protect your system....
Mar 02 09:24:47 d94c90c8f079 systemd[1]: Stopped Elastic Agent is a unified agent to observe, monitor and protect your system..
Mar 02 09:24:51 d94c90c8f079 systemd[1]: Started Elastic Agent is a unified agent to observe, monitor and protect your system..
Mar 02 09:25:17 d94c90c8f079 systemd[1]: Stopping Elastic Agent is a unified agent to observe, monitor and protect your system....
Mar 02 09:25:18 d94c90c8f079 systemd[1]: Stopped Elastic Agent is a unified agent to observe, monitor and protect your system..
Mar 02 09:38:20 d94c90c8f079 systemd[1]: Started Elastic Agent is a unified agent to observe, monitor and protect your system..
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.324Z	INFO	warn/warn.go:18	The Elastic Agent is currently in BETA and should not be used in production
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.324Z	INFO	application/application.go:65	Detecting execution mode
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.324Z	INFO	application/application.go:74	Agent is managed locally
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.324Z	INFO	capabilities/capabilities.go:59	capabilities file not found in /opt/Elastic/Agent/capabilities.yml
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.530Z	INFO	[composable]	composable/controller.go:44	EXPERIMENTAL - Inputs with variables are currently experimental and should not be used in production
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.532Z	INFO	[composable.providers.docker]	docker/docker.go:43	Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.532Z	INFO	[api]	api/server.go:62	Starting stats endpoint
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.532Z	INFO	application/local_mode.go:165	Agent is starting
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.534Z	INFO	application/local_mode.go:175	Agent is stopped
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.532Z	INFO	[api]	api/server.go:64	Metrics endpoint listening on: /tmp/elastic-agent/elastic-agent.sock (configured: unix:///tmp/elastic-agent/elastic-agent.sock)
Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.534Z	INFO	application/periodic.go:76	Configuration changes detected
Mar 02 09:38:21 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:21.306Z	INFO	stateresolver/stateresolver.go:48New State ID is 4qwgdKy-
Mar 02 09:38:21 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:21.306Z	INFO	stateresolver/stateresolver.go:49Converging state requires execution of 2 step(s)
Mar 02 09:38:21 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:21.306Z	INFO	operation/operation_fetch.go:61	metricbeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/metricbeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch
Mar 02 09:38:21 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:21.306Z	INFO	operation/operation_fetch.go:61	metricbeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/metricbeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch
Mar 02 09:38:22 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:22.336Z	INFO	process/app.go:174	Signaling application to stop because of shutdown: metricbeat--8.0.0-SNAPSHOT
```

**cat /opt/Elastic/Agent/data/elastic-agent-a84508/logs/elastic-agent-json.log**  (_using Fleet mode_)
```
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.374Z","log.origin":{"file.name":"warn/warn.go","file.line":18},"message":"The Elastic Agent is currently in BETA and should not be used in production","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.374Z","log.origin":{"file.name":"application/application.go","file.line":65},"message":"Detecting execution mode","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2021-03-02T09:38:22.374Z","log.logger":"cfgwarn","log.origin":{"file.name":"tlscommon/config.go","file.line":101},"message":"DEPRECATED: Treating the CommonName field on X.509 certificates as a host name when no Subject Alternative Names are present is going to be removed. Please update your certificates if needed. Will be removed in version: 8.0.0","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.374Z","log.origin":{"file.name":"application/application.go","file.line":90},"message":"Agent is managed by Fleet","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.375Z","log.origin":{"file.name":"capabilities/capabilities.go","file.line":59},"message":"capabilities file not found in /opt/Elastic/Agent/capabilities.yml","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.559Z","log.logger":"composable","log.origin":{"file.name":"composable/controller.go","file.line":44},"message":"EXPERIMENTAL - Inputs with variables are currently experimental and should not be used in production","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.560Z","log.logger":"composable.providers.docker","log.origin":{"file.name":"docker/docker.go","file.line":43},"message":"Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.561Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":62},"message":"Starting stats endpoint","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.561Z","log.origin":{"file.name":"application/managed_mode.go","file.line":273},"message":"Agent is starting","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2021-03-02T09:38:22.561Z","log.origin":{"file.name":"application/managed_mode.go","file.line":281},"message":"failed to ack update open /opt/Elastic/Agent/data/.update-marker: no such file or directory","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:22.561Z","log.logger":"api","log.origin":{"file.name":"api/server.go","file.line":64},"message":"Metrics endpoint listening on: /tmp/elastic-agent/elastic-agent.sock (configured: unix:///tmp/elastic-agent/elastic-agent.sock)","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:25.053Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":48},"message":"New State ID is lk8aU5US","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:25.053Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":49},"message":"Converging state requires execution of 3 step(s)","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:25.053Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"filebeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/filebeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:25.054Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"filebeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/filebeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:27.524Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:27Z: type: 'STATE': sub_type: 'STARTING' message: Application: filebeat--8.0.0-SNAPSHOT[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to STARTING: Starting","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:27.526Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"metricbeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/metricbeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:27.526Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"metricbeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/metricbeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:28.667Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:28Z: type: 'STATE': sub_type: 'RUNNING' message: Application: filebeat--8.0.0-SNAPSHOT[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to RUNNING: Running","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.522Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:29Z: type: 'STATE': sub_type: 'STARTING' message: Application: metricbeat--8.0.0-SNAPSHOT[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to STARTING: Starting","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.523Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"filebeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/filebeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.523Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"filebeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/filebeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.715Z","log.origin":{"file.name":"operation/operator.go","file.line":248},"message":"operation 'operation-install' skipped for filebeat.8.0.0-SNAPSHOT","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.885Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:29Z: type: 'STATE': sub_type: 'STARTING' message: Application: filebeat--8.0.0-SNAPSHOT--36643631373035623733363936343635[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to STARTING: Starting","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.888Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"metricbeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/metricbeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:29.888Z","log.origin":{"file.name":"operation/operation_fetch.go","file.line":61},"message":"metricbeat.8.0.0-SNAPSHOT already exists in /opt/Elastic/Agent/data/elastic-agent-a84508/downloads/metricbeat-8.0.0-SNAPSHOT-linux-x86_64.tar.gz. Skipping operation operation-fetch","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:30.058Z","log.origin":{"file.name":"operation/operator.go","file.line":248},"message":"operation 'operation-install' skipped for metricbeat.8.0.0-SNAPSHOT","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:30.139Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:30Z: type: 'STATE': sub_type: 'STARTING' message: Application: metricbeat--8.0.0-SNAPSHOT--36643631373035623733363936343635[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to STARTING: Starting","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:30.140Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":66},"message":"Updating internal state","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:30.920Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:30Z: type: 'STATE': sub_type: 'RUNNING' message: Application: metricbeat--8.0.0-SNAPSHOT[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to RUNNING: Running","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:31.069Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:31Z: type: 'STATE': sub_type: 'RUNNING' message: Application: filebeat--8.0.0-SNAPSHOT--36643631373035623733363936343635[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to RUNNING: Running","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-03-02T09:38:31.378Z","log.origin":{"file.name":"log/reporter.go","file.line":40},"message":"2021-03-02T09:38:31Z: type: 'STATE': sub_type: 'RUNNING' message: Application: metricbeat--8.0.0-SNAPSHOT--36643631373035623733363936343635[06ac8ab0-7b3b-11eb-92f2-a51b2e5e0225]: State changed to RUNNING: Running","ecs.version":"1.6.0"}
```

In the "same" millisecond:
```diff
- Mar 02 09:38:20 d94c90c8f079 elastic-agent[382]: 2021-03-02T09:38:20.324Z	INFO	application/application.go:74	Agent is managed locally
+ {"log.level":"info","@timestamp":"2021-03-02T09:38:22.374Z","log.origin":{"file.name":"application/application.go","file.line":90},"message":"Agent is managed by Fleet","ecs.version":"1.6.0"}
```

@michalpristas are my observations the expected behaviour?

Reply:
It seems fleet is not writing to journal, so we are going to stick to `catting` the file, even though there could be multiple rotated files